### Step 1. 필요한 라이브러리 임포트

In [1]:
import pandas as pd

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user) 확인

- user_id 유저 아이디
- age 나이
- gender 성별
- occupation 직업
- zip_code 우편번호

### Step 3. 

**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 users에 저장해주세요.**  
data : occupation.tsv

user_id를 인덱스로 설정해주세요.

In [4]:
users = pd.read_csv('C:/STUDY/StudyPandas/data/occupation.tsv',sep='|')
users

user_id  age gender     occupation zip_code
0          1   24      M     technician    85711
1          2   53      F          other    94043
2          3   23      M         writer    32067
3          4   24      M     technician    43537
4          5   33      F          other    15213
..       ...  ...    ...            ...      ...
938      939   26      F        student    33319
939      940   32      M  administrator    02215
940      941   20      M        student    97229
941      942   48      F      librarian    78209
942      943   22      M        student    77841

[943 rows x 5 columns]

### Step 4. 초반 25행 조회

In [ ]:
users.head(25)
#users.iloc[:25,:]

### Step 5. 마지막 10행 조회

In [ ]:
users.tail(10)
#users.iloc[-11:,:]

### Step 6. 데이터셋은 몇 개의 행으로 이루어져 있나요?

943

In [ ]:
users.shape[0]
#len(users)

### Step 7.  데이터셋은 몇 개의 열로 이루어져 있나요?

4

In [ ]:
users.shape[1]

### Step 8. 열(column)의 이름들을 모두 조회해주세요

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

In [ ]:
users.coulumns

### Step 9. 데이터셋의 인덱스를 조회해주세요

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

In [ ]:
users.index

### Step 10. 각 열의 데이터타입은 무엇인가요?

In [ ]:
users.dtypes

### Step 11. occupation column만 조회해주세요

In [ ]:
users['occupation']
# users.occupation

### Step 12. 데이터셋에는 몇 종류의 직업이 존재하나요?

21

In [ ]:
len(users['occupation'].unique())

### Step 13. 데이터셋에서 가장 많이 존재하는 직업은 무엇인가요?

student

In [ ]:
users['occupation'].value_counts()
# users.groupby('occupation')['age'].count().sort_values(ascending=False)

### Step 14. users.describe() 코드를 실행해서 숫자 열의 요약정보를 조회해주세요

In [ ]:
users.describe()

### Step 15. user들의 평균 나이는 몇 살인가요?
34

In [ ]:
round(users['age'].mean(),0)
# int(users['age'].mean())

### Step 16. 데이터셋에서 각각 여성과 남성 직업 순위 Top 5를 표시해주세요

* F  
occupation  
student          60  
administrator    36  
other            36  
librarian        29  
educator         26  
* M  
occupation  
student       136  
educator       69  
other          69  
engineer       65  
programmer     60  

In [ ]:
users.groupby('gender')['occupation'].value_counts().loc['F'].head()

In [ ]:
users.groupby('gender')['occupation'].value_counts().loc['M'].head()